### PR TITLE
[3.7] fix gauge assignment operators

### PR DESCRIPTION
**Pull request by kvahed**
### Scope & Purpose

*(Please describe the changes in this PR for reviewers - **mandatory**)*

- [X] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Backports:

- [X] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [X] This change is a trivial rework / code cleanup without any test coverage.
- [X] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [X] This PR adds tests that were used to verify all changes:
  - [X] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/12800/

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: tests/Metrics/MetricsTest.cpp [hunk 1]
```diff
@@ -257,6 +263,65 @@ template<typename T> void gauge_test() {
...
+  vector<T> vr(1000);
+  auto gen = [&dist, &mersenne_engine](){
+               return dist(mersenne_engine);
+             };
+  generate(vr.begin(), vr.end(), gen);
+
+  auto start = high_resolution_clock::now();
+  size_t const p = 10;
+  size_t const part = vr.size()/p;
+  std::vector<std::future<void>> f;
```
Comment (goedderz) [inline tests/Metrics/MetricsTest.cpp:281]: I guess you want the lambdas started as fast as possible, so:
```suggestion
  std::vector<std::future<void>> f;
  f.reserve(p);
```

File: tests/Metrics/MetricsTest.cpp [hunk 2]
```diff
@@ -257,6 +263,65 @@ template<typename T> void gauge_test() {
...
+                g *= vr.at(j);
+                g /= vr.at(j);
+              }}));
+  }
+  for (auto& i : f) {
+    i.wait();
+  }
+
+  if constexpr (std::is_same<T,float>::value) {
+    ASSERT_TRUE(1.f - g.load() < 1.e-6f);
```
Comment (goedderz) [inline tests/Metrics/MetricsTest.cpp:319]: gtests have a sane floating point equality built in:
```suggestion
    ASSERT_FLOAT_EQ(1.f, g.load());
```

That will also fix the missing `abs()` call. ;)

File: tests/Metrics/MetricsTest.cpp [hunk 3]
```diff
@@ -257,6 +263,65 @@ template<typename T> void gauge_test() {
...
+  } else {
+    ASSERT_TRUE(1. - g.load() < 1.e-15);
```
Comment (goedderz) [inline tests/Metrics/MetricsTest.cpp:321]:
```suggestion
    ASSERT_DOUBLE_EQ(1., g.load());
```

File: tests/Metrics/MetricsTest.cpp [hunk 4]
```diff
@@ -257,6 +263,65 @@ template<typename T> void gauge_test() {
...
+  g = one;
+  for (size_t i = 0; i < p; ++i) {
+    f.push_back(
+      async(launch::async,
+            [&] {
+              for (size_t j = 0; j < part; ++j) {
+                g += vr.at(j);
+                g -= vr.at(j);
+              }}));
+  }
```
Comment (goedderz) [inline tests/Metrics/MetricsTest.cpp:292]: Does that reproduce the bug reliably?

File: tests/Metrics/MetricsTest.cpp (hunk 2 above)
Comment (kvahed) [reply tests/Metrics/MetricsTest.cpp:320]: No, not reliable enough on windows

File: tests/Metrics/MetricsTest.cpp (hunk 3 above)
Comment (kvahed) [reply tests/Metrics/MetricsTest.cpp:322]: Again, not reliable on Windows. Tried this first

File: tests/Metrics/MetricsTest.cpp (hunk 4 above)
Comment (kvahed) [reply tests/Metrics/MetricsTest.cpp:293]: Yes, it does

**Review by neunhoef — APPROVED** [review]
LGTM

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12904/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-windows/6474/

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-mac-onlycatch/10249/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-windows/6477/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12911/

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12918/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**